osmosis-labs/osmosis
# Refactors testnetify and remove last localOsmosis Dockerfile

## What is the purpose of the change

As per the title, this PR:
1. refactors the `testnetify` scripts
2. removes the localOsmosis `Dockerfile`

Main changes:

- Give code a better structure
- Create a CLI tool with multiple arguments
- Decouple the `testnetify` process from the `osmosisd` binary. All the needed addresses are now expected as input.
- Reuse the main `Dockerfile` to avoid creating a dedicated one
- Significant speedup was obtained by:
    - reading/writing  from disk only at the beginning/end 
    - Replace the `sed_inplace` with the `replace` function that works in-memory
    - `IBC` module pruning
- Run the `testnetify.py` at `docker run` time and not at `docker build` time

## Brief Changelog

1.  Refactor `testnetify` scripts
2. Remove the localOsmosis `Dockerfile`

## Testing and Verifying

I have updated the documentation that can be followed to test the process.

Main parts:

1. Copy `state_export.json` in `tests/localosmosis/state_export/state_export.json` 
2. `localnet-build-state-export`
3. `localnet-start-state-export`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable
